Feature/make pip installable

rachmadaniHaryono opened the pull request:
i also move html_table_converter into the module. now it can be run with `html-table-converter` command (see `entry_points` on `setup.py`).

see the `setup` file, if there is any error on the entry.

the setup file generated first from pip-init, and checked with pyroma. 
some keywords taken from http://setuptools.readthedocs.io/en/latest/setuptools.html

readme change with some section taken from https://gist.github.com/PurpleBooth/109311bb0361f32d87a2

rachmadaniHaryono committed:
chg: dev: update html_table_converter

- move ifmain section to main function
- sort and move import module

rachmadaniHaryono committed:
new: usr: setup.py

rachmadaniHaryono committed:
chg: dev: update setup

- match from pyroma
- taken some keywords from http://setuptools.readthedocs.io/en/latest/setuptools.html
- fix entry_points

rachmadaniHaryono committed:
new: dev: tox config

rachmadaniHaryono committed:
new: dev: simple test

rachmadaniHaryono committed:
chg: dev: setup keyword based on init file

schmijos commented:
Hey @rachmadaniHaryono 

Thank you very much for your PR.
@ypid would you mind taking care of that? It seems to mainly affect the CLI.

schmijos commented on README.rst:
```diff
@@ -0,0 +1,85 @@
+Acknowledgments
+~~~~~~~~~~~~~~~
+
+- Hat tip to anyone who's code was used
```
I like your approach to better structure the README. Did you have a specific reason to use *.rst* instead or markdown?

ypid replied:
I welcome the switch to RST. It has many advantages over the not very well defined md format variants. Also, RST is very common in Python projects and PyPI which is probably why @rachmadaniHaryono changed it (see `setup.py` file).

rachmadaniHaryono replied:
@schmijos, to add what @ypid had said,  the related SO

https://stackoverflow.com/questions/26737222/pypi-description-markdown-doesnt-work

people can use pandoc to convert the markdown, but i will not risk on any error when converting and use rst directly

schmijos commented on README.rst:
```diff
@@ -0,0 +1,85 @@
+ # instantiate the parser and feed it
+ p = HTMLTableParser()
+ p.feed(xhtml)
+ pprint(p.tables)
```
I got rid of this part in my last edit of the README because I think people could look at the usage example in the code.

rachmadaniHaryono replied:
imo the benefit to put this on readme is actually to make user only read the source when he need to inspect the method or find the bug with the library. 

it also make people interested because people can see how simple the library is.

schmijos commented on README.rst:
```diff
@@ -0,0 +1,85 @@
+
+.. code:: bash
+
+ pip install git+git://github.com/schmijos/html-table-parser-python3.git
```
👍

ypid replied:
The code examples should to be indented in RST (ref: [directives](http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#directives)). Can you indent this and below code blocks to like so:

```RST
.. code:: bash

   pip install git+https://github.com/schmijos/html-table-parser-python3.git
```

Important: git:// is unencrypted/unauthenticated and should not be used.

rachmadaniHaryono replied:
afaik rst recognize the section even if the text indented with just one space

on this doc it mention several indentation on rst

http://docutils.sourceforge.net/docs/dev/rst/problems.html#section-structure-via-indentation

i will leave it @schmijos to choose which one and format it.

but i will change the pip install url to what  @ypid suggest

schmijos commented on README.rst:
```diff
@@ -0,0 +1,85 @@
+Running the tests
+~~~~~~~~~~~~~~~~~
+
+tba
```
Since you added a simple test, would you mind explaining what's a best practice to run them?

rachmadaniHaryono replied:
just added the test guide. but not quite sure if it is actually good enough.

with pyenv, just run pytest, or developer can use tox to run with available tox config.

schmijos commented on README.rst:
```diff
@@ -0,0 +1,85 @@
+Versioning
+~~~~~~~~~~
+
+We use SemVer for versioning. For the versions available, see the tags on this repository.
```
👍

ypid replied:
👍

schmijos commented on README.rst:
```diff
@@ -0,0 +1,85 @@
+
+- schmijos - Initial work
+
+See also the list of contributors who participated in this project.
```
I would leave this section away. It's already visible on the Github collaborators page and in the package description. Probably I also would get rid of it in the code itself since it's not part of the code.

rachmadaniHaryono replied:
>I would leave this section away. It's already visible on the Github collaborators page and in the package description. 

ok

>Probably I also would get rid of it in the code itself since it's not part of the code.

which part?

schmijos commented on tests/test_sample.py:
```diff
@@ -0,0 +1,6 @@
+
+def test_pass():
+    """test."""
+    assert True, "dummy sample test"
```
👍 Thats a start.

schmijos commented on tox.ini:
```diff
@@ -0,0 +1,6 @@
+
+[testenv]
+commands=py.test
+deps=pytest
```
I used *pyenv* for the env. Would you recommend to use tox and virtualenv?

rachmadaniHaryono replied:
pyenv and tox itself is different tools

i use both. pyenv is to test a single python version (at once), and tox to test it all at once.

e: will add test info on readme section.

ypid commented on README.rst:
```diff
@@ -0,0 +1,85 @@
+ from pprint import pprint
+ from html_table_parser import HTMLTableParser
+ 
+ target = 'http://www.twitter.com'
```
Can you do me a favor and update the legacy http URL scheme to state of the art HTTPS? Thanks 😉

ypid commented on setup.py:
```diff
@@ -0,0 +1,34 @@
+    # taken from http://setuptools.readthedocs.io/en/latest/setuptools.html
+    # see above for author, description and license
+    author_email="josua.schmid@renuo.ch",
+    keywords="hello world example examples",
```
`keywords` could be updated.

ypid reviewed (changes requested):
LGTM, Nice work @rachmadaniHaryono👍 A few comments are attached.

ypid commented:
> @ypid would you mind taking care of that? It seems to mainly affect the CLI.

I would like to, but I maintain enough projects already. @rachmadaniHaryono would you be interested? Things like publishing to PyPI, Sphinx docs, unit tests and all that good stuff seeks a maintainer :)

rachmadaniHaryono commented:
@ypid, @schmijos, not quite sure if can be maintainer.

i just found this library for my little project and add several improvement on a whim.

>Things like publishing to PyPI, Sphinx docs, unit tests and all that good stuff seeks a maintainer

i have no idea how to upload to pypi and sphinx docs. 

not quite sure sphinx docs is also needed due to how small the library is. even readme can handle the documentation for now. 

~~maybe i can add simple unittest, but no promise~~

simple test with simple html added

rachmadaniHaryono committed:
chg: dev: added updated readme.rst

- merge the usage section
- edit some words
- cli section
- remove authors section

rachmadaniHaryono committed:
chg: dev: keywords on setup.py

rachmadaniHaryono committed:
chg: test: test instruction.

rachmadaniHaryono committed:
new: test: simple table

ypid commented:
> i have no idea how to upload to pypi and sphinx docs.

Good opportunity to learn it. It is very nice stuff. I like to do this even for smaller projects because it gives the project a more professional look https://github.com/ypid/fdeunlock. It also helps with getting a [CII badge (example)](https://bestpractices.coreinfrastructure.org/projects/829).

rachmadaniHaryono committed:
chg: doc: use markdown

rachmadaniHaryono commented:
hi, i just made edit, so it is possible to use readme.md on pypi 

doc: https://dustingram.com/articles/2018/03/16/markdown-descriptions-on-pypi
example: https://test.pypi.org/project/html-table-parser-python3/

ahobsonsayers committed:
pip feature

Adding Files for publishing to PyPi
Create setup.py
Updated package name
Updated Readme and file structure
Updated version to match PyPi

rachmadaniHaryono commented:
hi @schmijos, i just review this pr once again and just found out that @ahobsonsayers also made changes and uploaded this package to pypi already.

imo i can rebase this pr once against @ahobsonsayers branch so his commit can also be added to this.

also @ahobsonsayers, can you give @schmijos and i access to pypi package so we can edit/upload

ahobsonsayers commented:
Hey @rachmadaniHaryono,

I must admit I didn't check the pull requests of this repository before uploading the package so i didn't see this conversation. If i had, id would have left a comment and opened a pull request following submitting to pypi.

Id be very happy to add you and @schmijos as owners of the package. Can I check, are your usernames the same as on github? Ill do that tomorrow if so.

rachmadaniHaryono commented:
>Id be very happy to add you and @schmijos as owners of the package. 

yeah, just add me as maintainer and @schmijos as owner

my name is the same with this one, but if you can't find it, here it is https://pypi.org/user/rachmadaniHaryono/

>I must admit I didn't check the pull requests of this repository before uploading the package so i didn't see this conversation. If i had, id would have left a comment and opened a pull request following submitting to pypi.

no problem, i will rebase this pr and use your commit as the base as much as possible. if there is difficulty or anything unclear i will ask you here.

rachmadaniHaryono committed:
chg: user: readme based on branch

rachmadaniHaryono committed:
Merge pull request #2 from rachmadaniHaryono/new-pip2

pip feature from @ahobsonsayers

rachmadaniHaryono commented:
i think i'm done. not too much different.

the setup.py use @ahobsonsayers work. also the entry on maintainer and author on that file is also changed. 

imo it can be used for next release @schmijos, because last release don't include entry point to html table converter command

ahobsonsayers commented:
@rachmadaniHaryono, ive added you as a maintainer now. @schmijos, i cannot find you on pypi - your username doesn't seem to exist. Can you let me know what it might be so I can add you as owner.

schmijos commented:
Hey guys,
I'm now on pypi as `schmijos`. Can I close this PR then?

rachmadaniHaryono commented:
i can't add you, i believe @ahobsonsayers (arranhs) can.

ypid commented:
Thanks for inviting me as collaborator also. Not sure what I can contribute yet.

ahobsonsayers commented:
Hey, sorry for the delay, my life got unbelievably complicated and I had to drop everything for a while.

@schmijos you are now an owner of the pypi package. @rachmadaniHaryono you are maintainer.

Again, sorry for the delay of 5 months.
